src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-form/rgw-bucket-form.component.html

```diff
@@ -410,6 +410,39 @@
           </div>
         </div>
 
+        <!-- Bucket policy -->
+        <legend class="cd-header"
+                i18n>Bucket policy
+        </legend>
+        <div class="row">
+          <div class="col-12">
+            <textarea #bucketPolicyTextArea
+                      class="form-control resize-vertical"
+                      id="bucket_policy"
+                      formControlName="bucket_policy"
+                      (change)="bucketPolicyOnChange()">
+            </textarea>
+            <span class="invalid-feedback"
+                  *ngIf="bucketForm.showError('bucket_policy', frm, 'invalidJson')"
+                  i18n>Invalid json text</span>
+            <button type="button"
+                    id="example-generator-button"
+                    class="btn btn-light float-end my-3"
+                    (click)="openUrl('https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-bucket-policies.html?icmpid=docs_amazons3_console')"
+                    i18n>
+              <i [ngClass]="[icons.externalUrl]"></i>
+              Policy examples
+            </button>
+            <button type="button"
+                    id="example-generator-button"
+                    class="btn btn-light float-end my-3"
+                    (click)="openUrl('https://awspolicygen.s3.amazonaws.com/policygen.html')"
+                    i18n>
+              <i [ngClass]="[icons.externalUrl]"></i>
+              Policy generator
+            </button>
```

Comment:
you can use a [button-group component](https://getbootstrap.com/docs/5.0/components/button-group/) to group these two together so it looks much more nicer

Reply:
Oh nice, added it!